test/apps/iodemo/io_demo.cc

```diff
@@ -1336,24 +1336,16 @@ class DemoClient : public P2pDemoCommon {
             _client->handle_operation_completion(_server_index, IO_READ,
                                                  _iov->data_size());
 
-            if (_status == UCS_OK) {
-                const server_info_t& server_info =
-                        _client->_server_info[_server_index];
-
-                ASSERTV(server_info.conn->ucx_status() == UCS_OK)
-                        << "conn_status=" << server_info.conn->ucx_status();
-
-                if (_validate) {
-                    validate(*_iov, _sn);
-                    if (_meta_comp_counter != 0) {
-                        // With tag API, we also wait for READ_COMP arrival, so need
-                        // to validate it. With AM API, READ_COMP arrives as AM
-                        // header together with data descriptor, we validate it in
-                        // place to avoid unneeded memory copy to this
-                        // IoReadResponseCallback _buffer.
-                        iomsg_t *msg = reinterpret_cast<iomsg_t*>(_buffer);
-                        validate(msg, _sn, _buffer_size);
-                    }
+            if ((_status == UCS_OK) && _validate) {
```

Comment:
should we validate only if _conn->ucx_status() == UCS_OK?

Reply:
according to our offline discussion, we shouldn't check the connection's status, since if some operation is completed with OK, the data should be valid